rgw/v2/tests/s3_swift/test_object_lock.py

```diff
@@ -0,0 +1,186 @@
+        rgw_conn = auth.do_auth()
+        # create buckets with object lock configuration
+        if config.test_ops["create_bucket"] is True:
+            log.info(f"no of buckets to create: {config.bucket_count}")
+            for bc in range(config.bucket_count):
+                bucket_name_to_create = utils.gen_bucket_name_from_userid(
+                    each_user["user_id"], rand_no=bc
+                )
+                log.info(f"creating bucket with name: {bucket_name_to_create}")
+                rgw_ip_and_port = get_rgw_ip_and_port()
+                s3_conn_client = boto3.client(
+                    "s3",
+                    aws_access_key_id=each_user["access_key"],
+                    aws_secret_access_key=each_user["secret_key"],
+                    endpoint_url=f"http://{rgw_ip_and_port}",
+                )
+                s3_conn_client.create_bucket(
+                    Bucket=bucket_name_to_create, ObjectLockEnabledForBucket=True
+                )
+                # put object lock configuration for bucket
+                s3_conn_client.put_object_lock_configuration(
+                    Bucket=bucket_name_to_create,
+                    ObjectLockConfiguration={
+                        "ObjectLockEnabled": "Enabled",
+                        "Rule": {"DefaultRetention": {"Mode": "COMPLIANCE", "Days": 1}},
+                    },
+                )
+                if config.test_ops["create_object"] is True:
+                    # uploading data
+                    log.info(f"s3 objects to create: {config.objects_count}")
+                    for oc, size in list(config.mapped_sizes.items()):
+                        s3_object_name = utils.gen_s3_object_name(
+                            bucket_name_to_create, 0
+                        )
+                        log.info(f"s3 object name: {s3_object_name}")
+                        s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+                        log.info(f"s3 object path: {s3_object_path}")
+                        log.info("upload type: normal")
+                        io_generator(TEST_DATA_PATH + "/" + s3_object_name, size)
+                        s3_conn_client.put_object(
+                            Body=TEST_DATA_PATH + "/" + s3_object_name,
+                            Bucket=bucket_name_to_create,
+                            Key=s3_object_name,
+                        )
+                    log.info("Verify version count")
+                    # Verify version count
+                    versions = s3_conn_client.list_object_versions(
+                        Bucket=bucket_name_to_create
+                    )
+                    versions_count = len(versions["Versions"])
+                    error_message = (
+                        f"Expected: {config.objects_count}, Actual: {versions_count}"
+                    )
+                    assert versions_count == config.objects_count, error_message
```

Comment:
don't use asserts anywhere in the test cases. exceptions make sense. please remove all the assertion. refer other tests.

Reply:
Modified it.